File: c++/20922.cpp
#include <algorithm>
#include <iostream>

using namespace std;

int arr[200001];
int countingArr[100001];

int main() {
  freopen("example.txt", "r", stdin);
  int N, K;
  cin >> N >> K;

  for (int i = 0; i < N; i++) {
    cin >> arr[i];
  }

  int i = 0;
  int j = 0;
  int ans = 1;
  countingArr[arr[i]] = 1;

  while (j < N && i <= j) {
    if (j + 1 < N && countingArr[arr[j + 1]] < K) {
      countingArr[arr[j + 1]]++;
      j++;
    } else {
      if (i == j)
        j++;
      else
        countingArr[arr[i]]--;
      i++;
    }

    ans = max(ans, j - i + 1);
  }

  cout << ans;

  return 0;
}

File: js/20922.js
const fs = require('fs');

const inputs = fs
  .readFileSync('example.txt')
  .toString()
  .trim()
  .split('\n')
  .map((e) => e.split(' ').map(Number));

const [N, K] = inputs[0];
const arr = inputs[1];
const counting = Array(100001).fill(0);

let i = 0;
let j = 0;
let ans = 1;

counting[arr[i]]++;

while (j < N && i <= j) {
  if (j + 1 < N && counting[arr[j + 1]] < K) {
    counting[arr[j + 1]]++;
    j++;
  } else {
    if (i === j) j++;
    else counting[arr[i]]--;
    i++;
  }

  ans = Math.max(ans, j - i + 1);
}

console.log(ans);
